drivers/wifi/simplelink/simplelink_sockets.c

```diff
@@ -317,39 +422,69 @@ static int simplelink_poll(struct pollfd *fds, int nfds, int msecs)
 #define SO_SNDBUF     (202)
 #define TCP_NODELAY   (203)
 
+/* Needed to keep line lengths < 80: */
+#define _SEC_DOMAIN_VERIF SL_SO_SECURE_DOMAIN_NAME_VERIFICATION
+
 static int simplelink_setsockopt(int sd, int level, int optname,
 				 const void *optval, socklen_t optlen)
 {
 	int retval;
 
-	/* Note: this logic should match SimpleLink SDK's socket.c: */
-	switch (optname) {
-	/* TCP_NODELAY is always set by the NWP; true always succeeds */
-	case TCP_NODELAY:
-		if (optval) {
-			/* if user wishes to have TCP_NODELAY = FALSE,
-			 * we return EINVAL and fail in the cases below.
-			 */
-			if (*(u32_t *)optval) {
-				retval = 0;
-				goto exit;
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS) && level == SOL_TLS) {
+		/* Handle Zephyr's SOL_TLS secure socket options: */
+		switch (optname) {
+		case TLS_SEC_TAG_LIST:
+			/* Bind credential filenames to this socket: */
+			retval = map_credentials(sd, optval, optlen);
+			break;
+		case TLS_HOSTNAME:
+			retval = sl_SetSockOpt(sd, SL_SOL_SOCKET,
+					       _SEC_DOMAIN_VERIF,
+					       (const char *)optval, optlen);
+			break;
+		case TLS_CIPHERSUITE_LIST:
+		case TLS_PEER_VERIFY:
+		case TLS_DTLS_ROLE:
+			/* Not yet supported: */
+			retval = ENOTSUP;
+			break;
+		default:
+			retval = EINVAL;
+			break;
+		}
+	} else {
+		/* Can be SOL_SOCKET or TI specific: */
+
+		/* Note: this logic should match SimpleLink SDK's socket.c: */
+		switch (optname) {
+		case TCP_NODELAY:
+			if (optval) {
+				/* if user wishes to have TCP_NODELAY = FALSE,
+				 * we return EINVAL and fail in the cases below.
+				 */
+				if (*(u32_t *)optval) {
+					retval = 0;
+					goto exit;
+				}
 			}
+			break;
```

Comment:
```
/* if user wishes to have TCP_NODELAY = FALSE,
 * we return EINVAL and fail in the cases below.
```
Just from above, this will not happen if we break here, right?

Reply:
Yes, look like the break is not wanted here.   Thanks for catching.